### ISSUE TITLE
'eqeqeq': ['error', 'smart'] not recognized

### ISSUE THREAD

**Issue by dotnetCarpenter** [eslint/eslint#9612]
**Tell us about your environment**

* **ESLint Version:** v4.10.0
* **Node Version:** v8.9.1
* **npm Version:** 5.5.1

**What parser (default, Babel-ESLint, etc.) are you using?** Babel-ESLint

<details>
<summary>Configuration</summary>

```js
module.exports = {
  root: true,
  parser: 'babel-eslint',
  env: {
    browser: true,
    node: true
  },
  extends: 'standard',
  // required to lint *.vue files
  plugins: [
    'html'
  ],
  // add your custom rules here
  rules: {
    'no-debugger': 0,
    'no-mixed-operators': 1,
    'eqeqeq': ['error', 'smart'] // <- 0 works
  },
  globals: {}
}
```
</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
export function range (start, step, stop) {
  if (stop == null) {
    stop = step
    step = stop - start
  }
  return gen(x => x, start, stop || step)
}
```

**Paste the command you used to run ESLint:**
N/A - it's run through [nuxt's](https://github.com/nuxt/nuxt.js/) default configuration.
But the output is: `68:30  error  Expected '===' and instead saw '=='  eqeqeq`.

**What did you expect to happen?**
According to [the docs](https://eslint.org/docs/rules/eqeqeq), using `eqeqeq: ["error", "smart"]` should *The "smart" option enforces the use of === and !== except for [...] Comparing against null*.

**What actually happened? Please include the actual, raw output from ESLint.**
`68:30  error  Expected '===' and instead saw '=='  eqeqeq`

**Comment by dotnetCarpenter** [comment]
I should make it clear, that setting the value to `0` works, so it is picked up and I can turn off the rule. Unfortunately, it doesn't pick up the array value and it's effectively turned on always.

**Comment by platinumazure** [comment]
@dotnetCarpenter I can't recreate in our [online demo](https://eslint.org/demo/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXVuZGVmOiBvZmYsIGVxZXFlcTogW2Vycm9yLCBzbWFydF0gKi9cclxuZXhwb3J0IGZ1bmN0aW9uIHJhbmdlIChzdGFydCwgc3RlcCwgc3RvcCkge1xyXG4gIGlmIChzdG9wID09IG51bGwpIHtcclxuICAgIHN0b3AgPSBzdGVwXHJcbiAgICBzdGVwID0gc3RvcCAtIHN0YXJ0XHJcbiAgfVxyXG4gIHJldHVybiBnZW4oeCA9PiB4LCBzdGFydCwgc3RvcCB8fCBzdGVwKVxyXG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo1LCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7ImpzeCI6dHJ1ZX19LCJydWxlcyI6eyJjb25zdHJ1Y3Rvci1zdXBlciI6Miwibm8tY2FzZS1kZWNsYXJhdGlvbnMiOjIsIm5vLWNsYXNzLWFzc2lnbiI6Miwibm8tY29tcGFyZS1uZWctemVybyI6Miwibm8tY29uZC1hc3NpZ24iOjIsIm5vLWNvbnNvbGUiOjIsIm5vLWNvbnN0LWFzc2lnbiI6Miwibm8tY29uc3RhbnQtY29uZGl0aW9uIjoyLCJuby1jb250cm9sLXJlZ2V4IjoyLCJuby1kZWJ1Z2dlciI6Miwibm8tZGVsZXRlLXZhciI6Miwibm8tZHVwZS1hcmdzIjoyLCJuby1kdXBlLWNsYXNzLW1lbWJlcnMiOjIsIm5vLWR1cGUta2V5cyI6Miwibm8tZHVwbGljYXRlLWNhc2UiOjIsIm5vLWVtcHR5LWNoYXJhY3Rlci1jbGFzcyI6Miwibm8tZW1wdHktcGF0dGVybiI6Miwibm8tZW1wdHkiOjIsIm5vLWV4LWFzc2lnbiI6Miwibm8tZXh0cmEtYm9vbGVhbi1jYXN0IjoyLCJuby1leHRyYS1zZW1pIjoyLCJuby1mYWxsdGhyb3VnaCI6Miwibm8tZnVuYy1hc3NpZ24iOjIsIm5vLWdsb2JhbC1hc3NpZ24iOjIsIm5vLWlubmVyLWRlY2xhcmF0aW9ucyI6Miwibm8taW52YWxpZC1yZWdleHAiOjIsIm5vLWlycmVndWxhci13aGl0ZXNwYWNlIjoyLCJuby1taXhlZC1zcGFjZXMtYW5kLXRhYnMiOjIsIm5vLW5ldy1zeW1ib2wiOjIsIm5vLW9iai1jYWxscyI6Miwibm8tb2N0YWwiOjIsIm5vLXJlZGVjbGFyZSI6Miwibm8tcmVnZXgtc3BhY2VzIjoyLCJuby1zZWxmLWFzc2lnbiI6Miwibm8tc3BhcnNlLWFycmF5cyI6Miwibm8tdGhpcy1iZWZvcmUtc3VwZXIiOjIsIm5vLXVuZGVmIjoyLCJuby11bmV4cGVjdGVkLW11bHRpbGluZSI6Miwibm8tdW5yZWFjaGFibGUiOjIsIm5vLXVuc2FmZS1maW5hbGx5IjoyLCJuby11bnNhZmUtbmVnYXRpb24iOjIsIm5vLXVudXNlZC1sYWJlbHMiOjIsIm5vLXVudXNlZC12YXJzIjoyLCJuby11c2VsZXNzLWVzY2FwZSI6MiwicmVxdWlyZS15aWVsZCI6MiwidXNlLWlzbmFuIjoyLCJ2YWxpZC10eXBlb2YiOjJ9LCJlbnYiOnt9fX0=).

Can you please try running the ESLint CLI and see if you can still recreate the error? And if so, could you run `eslint --print-config path/to/file.js` and see what the eqeqeq setting looks like there? In both of these cases, we might see if there is a configuration somewhere else that is overriding what you have specified.

**Comment by dotnetCarpenter** [comment]
Sure!
`node_modules/.bin/eslint assets/js/utils.js`
```
  68:30  error  Expected '===' and instead saw '=='  eqeqeq

✖ 1 problem (1 error, 0 warnings)
```

`node_modules/.bin/eslint --print-config assets/js/utils.js`:

<details>
<summary>Output</summary>

```json
{
  "globals": {
    "Array": false,
    "ArrayBuffer": false,
    "Boolean": false,
    "constructor": false,
    "DataView": false,
    "Date": false,
    "decodeURI": false,
    "decodeURIComponent": false,
    "encodeURI": false,
    "encodeURIComponent": false,
    "Error": false,
    "escape": false,
    "eval": false,
    "EvalError": false,
    "Float32Array": false,
    "Float64Array": false,
    "Function": false,
    "hasOwnProperty": false,
    "Infinity": false,
    "Int16Array": false,
    "Int32Array": false,
    "Int8Array": false,
    "isFinite": false,
    "isNaN": false,
    "isPrototypeOf": false,
    "JSON": false,
    "Map": false,
    "Math": false,
    "NaN": false,
    "Number": false,
    "Object": false,
    "parseFloat": false,
    "parseInt": false,
    "Promise": false,
    "propertyIsEnumerable": false,
    "Proxy": false,
    "RangeError": false,
    "ReferenceError": false,
    "Reflect": false,
    "RegExp": false,
    "Set": false,
    "String": false,
    "Symbol": false,
    "SyntaxError": false,
    "System": false,
    "toLocaleString": false,
    "toString": false,
    "TypeError": false,
    "Uint16Array": false,
    "Uint32Array": false,
    "Uint8Array": false,
    "Uint8ClampedArray": false,
    "undefined": false,
    "unescape": false,
    "URIError": false,
    "valueOf": false,
    "WeakMap": false,
    "WeakSet": false,
    "__dirname": false,
    "__filename": false,
    "arguments": false,
    "Buffer": false,
    "clearImmediate": false,
    "clearInterval": false,
    "clearTimeout": false,
    "console": false,
    "exports": true,
    "GLOBAL": false,
    "global": false,
    "Intl": false,
    "module": false,
    "process": false,
    "require": false,
    "root": false,
    "setImmediate": false,
    "setInterval": false,
    "setTimeout": false,
    "addEventListener": false,
    "alert": false,
    "AnalyserNode": false,
    "Animation": false,
    "AnimationEffectReadOnly": false,
    "AnimationEffectTiming": false,
    "AnimationEffectTimingReadOnly": false,
    "AnimationEvent": false,
    "AnimationPlaybackEvent": false,
    "AnimationTimeline": false,
    "applicationCache": false,
    "ApplicationCache": false,
    "ApplicationCacheErrorEvent": false,
    "atob": false,
    "Attr": false,
    "Audio": false,
    "AudioBuffer": false,
    "AudioBufferSourceNode": false,
    "AudioContext": false,
    "AudioDestinationNode": false,
    "AudioListener": false,
    "AudioNode": false,
    "AudioParam": false,
    "AudioProcessingEvent": false,
    "AutocompleteErrorEvent": false,
    "BarProp": false,
    "BatteryManager": false,
    "BeforeUnloadEvent": false,
    "BiquadFilterNode": false,
    "Blob": false,
    "blur": false,
    "btoa": false,
    "Cache": false,
    "caches": false,
    "CacheStorage": false,
    "cancelAnimationFrame": false,
    "cancelIdleCallback": false,
    "CanvasGradient": false,
    "CanvasPattern": false,
    "CanvasRenderingContext2D": false,
    "CDATASection": false,
    "ChannelMergerNode": false,
    "ChannelSplitterNode": false,
    "CharacterData": false,
    "clientInformation": false,
    "ClientRect": false,
    "ClientRectList": false,
    "ClipboardEvent": false,
    "close": false,
    "closed": false,
    "CloseEvent": false,
    "Comment": false,
    "CompositionEvent": false,
    "confirm": false,
    "ConvolverNode": false,
    "createImageBitmap": false,
    "Credential": false,
    "CredentialsContainer": false,
    "crypto": false,
    "Crypto": false,
    "CryptoKey": false,
    "CSS": false,
    "CSSAnimation": false,
    "CSSFontFaceRule": false,
    "CSSImportRule": false,
    "CSSKeyframeRule": false,
    "CSSKeyframesRule": false,
    "CSSMediaRule": false,
    "CSSPageRule": false,
    "CSSRule": false,
    "CSSRuleList": false,
    "CSSStyleDeclaration": false,
    "CSSStyleRule": false,
    "CSSStyleSheet": false,
    "CSSSupportsRule": false,
    "CSSTransition": false,
    "CSSUnknownRule": false,
    "CSSViewportRule": false,
    "customElements": false,
    "CustomEvent": false,
    "DataTransfer": false,
    "DataTransferItem": false,
    "DataTransferItemList": false,
    "Debug": false,
    "defaultStatus": false,
    "defaultstatus": false,
    "DelayNode": false,
    "DeviceMotionEvent": false,
    "DeviceOrientationEvent": false,
    "devicePixelRatio": false,
    "dispatchEvent": false,
    "document": false,
    "Document": false,
    "DocumentFragment": false,
    "DocumentTimeline": false,
    "DocumentType": false,
    "DOMError": false,
    "DOMException": false,
    "DOMImplementation": false,
    "DOMParser": false,
    "DOMSettableTokenList": false,
    "DOMStringList": false,
    "DOMStringMap": false,
    "DOMTokenList": false,
    "DragEvent": false,
    "DynamicsCompressorNode": false,
    "Element": false,
    "ElementTimeControl": false,
    "ErrorEvent": false,
    "event": false,
    "Event": false,
    "EventSource": false,
    "EventTarget": false,
    "external": false,
    "FederatedCredential": false,
    "fetch": false,
    "File": false,
    "FileError": false,
    "FileList": false,
    "FileReader": false,
    "find": false,
    "focus": false,
    "FocusEvent": false,
    "FontFace": false,
    "FormData": false,
    "frameElement": false,
    "frames": false,
    "GainNode": false,
    "Gamepad": false,
    "GamepadButton": false,
    "GamepadEvent": false,
    "getComputedStyle": false,
    "getSelection": false,
    "HashChangeEvent": false,
    "Headers": false,
    "history": false,
    "History": false,
    "HTMLAllCollection": false,
    "HTMLAnchorElement": false,
    "HTMLAppletElement": false,
    "HTMLAreaElement": false,
    "HTMLAudioElement": false,
    "HTMLBaseElement": false,
    "HTMLBlockquoteElement": false,
    "HTMLBodyElement": false,
    "HTMLBRElement": false,
    "HTMLButtonElement": false,
    "HTMLCanvasElement": false,
    "HTMLCollection": false,
    "HTMLContentElement": false,
    "HTMLDataListElement": false,
    "HTMLDetailsElement": false,
    "HTMLDialogElement": false,
    "HTMLDirectoryElement": false,
    "HTMLDivElement": false,
    "HTMLDListElement": false,
    "HTMLDocument": false,
    "HTMLElement": false,
    "HTMLEmbedElement": false,
    "HTMLFieldSetElement": false,
    "HTMLFontElement": false,
    "HTMLFormControlsCollection": false,
    "HTMLFormElement": false,
    "HTMLFrameElement": false,
    "HTMLFrameSetElement": false,
    "HTMLHeadElement": false,
    "HTMLHeadingElement": false,
    "HTMLHRElement": false,
    "HTMLHtmlElement": false,
    "HTMLIFrameElement": false,
    "HTMLImageElement": false,
    "HTMLInputElement": false,
    "HTMLIsIndexElement": false,
    "HTMLKeygenElement": false,
    "HTMLLabelElement": false,
    "HTMLLayerElement": false,
    "HTMLLegendElement": false,
    "HTMLLIElement": false,
    "HTMLLinkElement": false,
    "HTMLMapElement": false,
    "HTMLMarqueeElement": false,
    "HTMLMediaElement": false,
    "HTMLMenuElement": false,
    "HTMLMetaElement": false,
    "HTMLMeterElement": false,
    "HTMLModElement": false,
    "HTMLObjectElement": false,
    "HTMLOListElement": false,
    "HTMLOptGroupElement": false,
    "HTMLOptionElement": false,
    "HTMLOptionsCollection": false,
    "HTMLOutputElement": false,
    "HTMLParagraphElement": false,
    "HTMLParamElement": false,
    "HTMLPictureElement": false,
    "HTMLPreElement": false,
    "HTMLProgressElement": false,
    "HTMLQuoteElement": false,
    "HTMLScriptElement": false,
    "HTMLSelectElement": false,
    "HTMLShadowElement": false,
    "HTMLSourceElement": false,
    "HTMLSpanElement": false,
    "HTMLStyleElement": false,
    "HTMLTableCaptionElement": false,
    "HTMLTableCellElement": false,
    "HTMLTableColElement": false,
    "HTMLTableElement": false,
    "HTMLTableRowElement": false,
    "HTMLTableSectionElement": false,
    "HTMLTemplateElement": false,
    "HTMLTextAreaElement": false,
    "HTMLTitleElement": false,
    "HTMLTrackElement": false,
    "HTMLUListElement": false,
    "HTMLUnknownElement": false,
    "HTMLVideoElement": false,
    "IDBCursor": false,
    "IDBCursorWithValue": false,
    "IDBDatabase": false,
    "IDBEnvironment": false,
    "IDBFactory": false,
    "IDBIndex": false,
    "IDBKeyRange": false,
    "IDBObjectStore": false,
    "IDBOpenDBRequest": false,
    "IDBRequest": false,
    "IDBTransaction": false,
    "IDBVersionChangeEvent": false,
    "Image": false,
    "ImageBitmap": false,
    "ImageData": false,
    "indexedDB": false,
    "innerHeight": false,
    "innerWidth": false,
    "InputEvent": false,
    "InputMethodContext": false,
    "IntersectionObserver": false,
    "IntersectionObserverEntry": false,
    "KeyboardEvent": false,
    "KeyframeEffect": false,
    "KeyframeEffectReadOnly": false,
    "length": false,
    "localStorage": false,
    "location": false,
    "Location": false,
    "locationbar": false,
    "matchMedia": false,
    "MediaElementAudioSourceNode": false,
    "MediaEncryptedEvent": false,
    "MediaError": false,
    "MediaKeyError": false,
    "MediaKeyEvent": false,
    "MediaKeyMessageEvent": false,
    "MediaKeys": false,
    "MediaKeySession": false,
    "MediaKeyStatusMap": false,
    "MediaKeySystemAccess": false,
    "MediaList": false,
    "MediaQueryList": false,
    "MediaQueryListEvent": false,
    "MediaSource": false,
    "MediaRecorder": false,
    "MediaStream": false,
    "MediaStreamAudioDestinationNode": false,
    "MediaStreamAudioSourceNode": false,
    "MediaStreamEvent": false,
    "MediaStreamTrack": false,
    "menubar": false,
    "MessageChannel": false,
    "MessageEvent": false,
    "MessagePort": false,
    "MIDIAccess": false,
    "MIDIConnectionEvent": false,
    "MIDIInput": false,
    "MIDIInputMap": false,
    "MIDIMessageEvent": false,
    "MIDIOutput": false,
    "MIDIOutputMap": false,
    "MIDIPort": false,
    "MimeType": false,
    "MimeTypeArray": false,
    "MouseEvent": false,
    "moveBy": false,
    "moveTo": false,
    "MutationEvent": false,
    "MutationObserver": false,
    "MutationRecord": false,
    "name": false,
    "NamedNodeMap": false,
    "navigator": false,
    "Navigator": false,
    "Node": false,
    "NodeFilter": false,
    "NodeIterator": false,
    "NodeList": false,
    "Notification": false,
    "OfflineAudioCompletionEvent": false,
    "OfflineAudioContext": false,
    "offscreenBuffering": false,
    "onbeforeunload": true,
    "onblur": true,
    "onerror": true,
    "onfocus": true,
    "onload": true,
    "onresize": true,
    "onunload": true,
    "open": false,
    "openDatabase": false,
    "opener": false,
    "opera": false,
    "Option": false,
    "OscillatorNode": false,
    "outerHeight": false,
    "outerWidth": false,
    "PageTransitionEvent": false,
    "pageXOffset": false,
    "pageYOffset": false,
    "parent": false,
    "PasswordCredential": false,
    "Path2D": false,
    "performance": false,
    "Performance": false,
    "PerformanceEntry": false,
    "PerformanceMark": false,
    "PerformanceMeasure": false,
    "PerformanceNavigation": false,
    "PerformanceResourceTiming": false,
    "PerformanceTiming": false,
    "PeriodicWave": false,
    "Permissions": false,
    "PermissionStatus": false,
    "personalbar": false,
    "Plugin": false,
    "PluginArray": false,
    "PopStateEvent": false,
    "postMessage": false,
    "print": false,
    "ProcessingInstruction": false,
    "ProgressEvent": false,
    "PromiseRejectionEvent": false,
    "prompt": false,
    "PushManager": false,
    "PushSubscription": false,
    "RadioNodeList": false,
    "Range": false,
    "ReadableByteStream": false,
    "ReadableStream": false,
    "removeEventListener": false,
    "Request": false,
    "requestAnimationFrame": false,
    "requestIdleCallback": false,
    "resizeBy": false,
    "resizeTo": false,
    "Response": false,
    "RTCIceCandidate": false,
    "RTCSessionDescription": false,
    "RTCPeerConnection": false,
    "screen": false,
    "Screen": false,
    "screenLeft": false,
    "ScreenOrientation": false,
    "screenTop": false,
    "screenX": false,
    "screenY": false,
    "ScriptProcessorNode": false,
    "scroll": false,
    "scrollbars": false,
    "scrollBy": false,
    "scrollTo": false,
    "scrollX": false,
    "scrollY": false,
    "SecurityPolicyViolationEvent": false,
    "Selection": false,
    "self": false,
    "ServiceWorker": false,
    "ServiceWorkerContainer": false,
    "ServiceWorkerRegistration": false,
    "sessionStorage": false,
    "ShadowRoot": false,
    "SharedKeyframeList": false,
    "SharedWorker": false,
    "showModalDialog": false,
    "SiteBoundCredential": false,
    "speechSynthesis": false,
    "SpeechSynthesisEvent": false,
    "SpeechSynthesisUtterance": false,
    "status": false,
    "statusbar": false,
    "stop": false,
    "Storage": false,
    "StorageEvent": false,
    "styleMedia": false,
    "StyleSheet": false,
    "StyleSheetList": false,
    "SubtleCrypto": false,
    "SVGAElement": false,
    "SVGAltGlyphDefElement": false,
    "SVGAltGlyphElement": false,
    "SVGAltGlyphItemElement": false,
    "SVGAngle": false,
    "SVGAnimateColorElement": false,
    "SVGAnimatedAngle": false,
    "SVGAnimatedBoolean": false,
    "SVGAnimatedEnumeration": false,
    "SVGAnimatedInteger": false,
    "SVGAnimatedLength": false,
    "SVGAnimatedLengthList": false,
    "SVGAnimatedNumber": false,
    "SVGAnimatedNumberList": false,
    "SVGAnimatedPathData": false,
    "SVGAnimatedPoints": false,
    "SVGAnimatedPreserveAspectRatio": false,
    "SVGAnimatedRect": false,
    "SVGAnimatedString": false,
    "SVGAnimatedTransformList": false,
    "SVGAnimateElement": false,
    "SVGAnimateMotionElement": false,
    "SVGAnimateTransformElement": false,
    "SVGAnimationElement": false,
    "SVGCircleElement": false,
    "SVGClipPathElement": false,
    "SVGColor": false,
    "SVGColorProfileElement": false,
    "SVGColorProfileRule": false,
    "SVGComponentTransferFunctionElement": false,
    "SVGCSSRule": false,
    "SVGCursorElement": false,
    "SVGDefsElement": false,
    "SVGDescElement": false,
    "SVGDiscardElement": false,
    "SVGDocument": false,
    "SVGElement": false,
    "SVGElementInstance": false,
    "SVGElementInstanceList": false,
    "SVGEllipseElement": false,
    "SVGEvent": false,
    "SVGExternalResourcesRequired": false,
    "SVGFEBlendElement": false,
    "SVGFEColorMatrixElement": false,
    "SVGFEComponentTransferElement": false,
    "SVGFECompositeElement": false,
    "SVGFEConvolveMatrixElement": false,
    "SVGFEDiffuseLightingElement": false,
    "SVGFEDisplacementMapElement": false,
    "SVGFEDistantLightElement": false,
    "SVGFEDropShadowElement": false,
    "SVGFEFloodElement": false,
    "SVGFEFuncAElement": false,
    "SVGFEFuncBElement": false,
    "SVGFEFuncGElement": false,
    "SVGFEFuncRElement": false,
    "SVGFEGaussianBlurElement": false,
    "SVGFEImageElement": false,
    "SVGFEMergeElement": false,
    "SVGFEMergeNodeElement": false,
    "SVGFEMorphologyElement": false,
    "SVGFEOffsetElement": false,
    "SVGFEPointLightElement": false,
    "SVGFESpecularLightingElement": false,
    "SVGFESpotLightElement": false,
    "SVGFETileElement": false,
    "SVGFETurbulenceElement": false,
    "SVGFilterElement": false,
    "SVGFilterPrimitiveStandardAttributes": false,
    "SVGFitToViewBox": false,
    "SVGFontElement": false,
    "SVGFontFaceElement": false,
    "SVGFontFaceFormatElement": false,
    "SVGFontFaceNameElement": false,
    "SVGFontFaceSrcElement": false,
    "SVGFontFaceUriElement": false,
    "SVGForeignObjectElement": false,
    "SVGGElement": false,
    "SVGGeometryElement": false,
    "SVGGlyphElement": false,
    "SVGGlyphRefElement": false,
    "SVGGradientElement": false,
    "SVGGraphicsElement": false,
    "SVGHKernElement": false,
    "SVGICCColor": false,
    "SVGImageElement": false,
    "SVGLangSpace": false,
    "SVGLength": false,
    "SVGLengthList": false,
    "SVGLinearGradientElement": false,
    "SVGLineElement": false,
    "SVGLocatable": false,
    "SVGMarkerElement": false,
    "SVGMaskElement": false,
    "SVGMatrix": false,
    "SVGMetadataElement": false,
    "SVGMissingGlyphElement": false,
    "SVGMPathElement": false,
    "SVGNumber": false,
    "SVGNumberList": false,
    "SVGPaint": false,
    "SVGPathElement": false,
    "SVGPathSeg": false,
    "SVGPathSegArcAbs": false,
    "SVGPathSegArcRel": false,
    "SVGPathSegClosePath": false,
    "SVGPathSegCurvetoCubicAbs": false,
    "SVGPathSegCurvetoCubicRel": false,
    "SVGPathSegCurvetoCubicSmoothAbs": false,
    "SVGPathSegCurvetoCubicSmoothRel": false,
    "SVGPathSegCurvetoQuadraticAbs": false,
    "SVGPathSegCurvetoQuadraticRel": false,
    "SVGPathSegCurvetoQuadraticSmoothAbs": false,
    "SVGPathSegCurvetoQuadraticSmoothRel": false,
    "SVGPathSegLinetoAbs": false,
    "SVGPathSegLinetoHorizontalAbs": false,
    "SVGPathSegLinetoHorizontalRel": false,
    "SVGPathSegLinetoRel": false,
    "SVGPathSegLinetoVerticalAbs": false,
    "SVGPathSegLinetoVerticalRel": false,
    "SVGPathSegList": false,
    "SVGPathSegMovetoAbs": false,
    "SVGPathSegMovetoRel": false,
    "SVGPatternElement": false,
    "SVGPoint": false,
    "SVGPointList": false,
    "SVGPolygonElement": false,
    "SVGPolylineElement": false,
    "SVGPreserveAspectRatio": false,
    "SVGRadialGradientElement": false,
    "SVGRect": false,
    "SVGRectElement": false,
    "SVGRenderingIntent": false,
    "SVGScriptElement": false,
    "SVGSetElement": false,
    "SVGStopElement": false,
    "SVGStringList": false,
    "SVGStylable": false,
    "SVGStyleElement": false,
    "SVGSVGElement": false,
    "SVGSwitchElement": false,
    "SVGSymbolElement": false,
    "SVGTests": false,
    "SVGTextContentElement": false,
    "SVGTextElement": false,
    "SVGTextPathElement": false,
    "SVGTextPositioningElement": false,
    "SVGTitleElement": false,
    "SVGTransform": false,
    "SVGTransformable": false,
    "SVGTransformList": false,
    "SVGTRefElement": false,
    "SVGTSpanElement": false,
    "SVGUnitTypes": false,
    "SVGURIReference": false,
    "SVGUseElement": false,
    "SVGViewElement": false,
    "SVGViewSpec": false,
    "SVGVKernElement": false,
    "SVGZoomAndPan": false,
    "SVGZoomEvent": false,
    "Text": false,
    "TextDecoder": false,
    "TextEncoder": false,
    "TextEvent": false,
    "TextMetrics": false,
    "TextTrack": false,
    "TextTrackCue": false,
    "TextTrackCueList": false,
    "TextTrackList": false,
    "TimeEvent": false,
    "TimeRanges": false,
    "toolbar": false,
    "top": false,
    "Touch": false,
    "TouchEvent": false,
    "TouchList": false,
    "TrackEvent": false,
    "TransitionEvent": false,
    "TreeWalker": false,
    "UIEvent": false,
    "URL": false,
    "URLSearchParams": false,
    "ValidityState": false,
    "VTTCue": false,
    "WaveShaperNode": false,
    "WebGLActiveInfo": false,
    "WebGLBuffer": false,
    "WebGLContextEvent": false,
    "WebGLFramebuffer": false,
    "WebGLProgram": false,
    "WebGLRenderbuffer": false,
    "WebGLRenderingContext": false,
    "WebGLShader": false,
    "WebGLShaderPrecisionFormat": false,
    "WebGLTexture": false,
    "WebGLUniformLocation": false,
    "WebSocket": false,
    "WheelEvent": false,
    "window": false,
    "Window": false,
    "Worker": false,
    "XDomainRequest": false,
    "XMLDocument": false,
    "XMLHttpRequest": false,
    "XMLHttpRequestEventTarget": false,
    "XMLHttpRequestProgressEvent": false,
    "XMLHttpRequestUpload": false,
    "XMLSerializer": false,
    "XPathEvaluator": false,
    "XPathException": false,
    "XPathExpression": false,
    "XPathNamespace": false,
    "XPathNSResolver": false,
    "XPathResult": false,
    "XSLTProcessor": false
  },
  "env": {
    "es6": true,
    "node": true,
    "browser": true
  },
  "rules": {
    "accessor-pairs": "error",
    "arrow-spacing": [
      "error",
      {
        "before": true,
        "after": true
      }
    ],
    "block-spacing": [
      "error",
      "always"
    ],
    "brace-style": [
      "error",
      "1tbs",
      {
        "allowSingleLine": true
      }
    ],
    "camelcase": [
      "error",
      {
        "properties": "never"
      }
    ],
    "comma-dangle": [
      "error",
      {
        "arrays": "never",
        "objects": "never",
        "imports": "never",
        "exports": "never",
        "functions": "never"
      }
    ],
    "comma-spacing": [
      "error",
      {
        "before": false,
        "after": true
      }
    ],
    "comma-style": [
      "error",
      "last"
    ],
    "constructor-super": "error",
    "curly": [
      "error",
      "multi-line"
    ],
    "dot-location": [
      "error",
      "property"
    ],
    "eol-last": "error",
    "eqeqeq": [
      "error",
      "smart"
    ],
    "func-call-spacing": [
      "error",
      "never"
    ],
    "generator-star-spacing": [
      "error",
      {
        "before": true,
        "after": true
      }
    ],
    "handle-callback-err": [
      "error",
      "^(err|error)$"
    ],
    "indent": [
      "error",
      2,
      {
        "SwitchCase": 1
      }
    ],
    "key-spacing": [
      "error",
      {
        "beforeColon": false,
        "afterColon": true
      }
    ],
    "keyword-spacing": [
      "error",
      {
        "before": true,
        "after": true
      }
    ],
    "new-cap": [
      "error",
      {
        "newIsCap": true,
        "capIsNew": false
      }
    ],
    "new-parens": "error",
    "no-array-constructor": "error",
    "no-caller": "error",
    "no-class-assign": "error",
    "no-compare-neg-zero": "error",
    "no-cond-assign": "error",
    "no-const-assign": "error",
    "no-constant-condition": [
      "error",
      {
        "checkLoops": false
      }
    ],
    "no-control-regex": "error",
    "no-debugger": 0,
    "no-delete-var": "error",
    "no-dupe-args": "error",
    "no-dupe-class-members": "error",
    "no-dupe-keys": "error",
    "no-duplicate-case": "error",
    "no-empty-character-class": "error",
    "no-empty-pattern": "error",
    "no-eval": "error",
    "no-ex-assign": "error",
    "no-extend-native": "error",
    "no-extra-bind": "error",
    "no-extra-boolean-cast": "error",
    "no-extra-parens": [
      "error",
      "functions"
    ],
    "no-fallthrough": "error",
    "no-floating-decimal": "error",
    "no-func-assign": "error",
    "no-global-assign": "error",
    "no-implied-eval": "error",
    "no-inner-declarations": [
      "error",
      "functions"
    ],
    "no-invalid-regexp": "error",
    "no-irregular-whitespace": "error",
    "no-iterator": "error",
    "no-label-var": "error",
    "no-labels": [
      "error",
      {
        "allowLoop": false,
        "allowSwitch": false
      }
    ],
    "no-lone-blocks": "error",
    "no-mixed-operators": [
      1,
      {
        "groups": [
          [
            "==",
            "!=",
            "===",
            "!==",
            ">",
            ">=",
            "<",
            "<="
          ],
          [
            "&&",
            "||"
          ],
          [
            "in",
            "instanceof"
          ]
        ],
        "allowSamePrecedence": true
      }
    ],
    "no-mixed-spaces-and-tabs": "error",
    "no-multi-spaces": "error",
    "no-multi-str": "error",
    "no-multiple-empty-lines": [
      "error",
      {
        "max": 1,
        "maxEOF": 0
      }
    ],
    "no-negated-in-lhs": "error",
    "no-new": "error",
    "no-new-func": "error",
    "no-new-object": "error",
    "no-new-require": "error",
    "no-new-symbol": "error",
    "no-new-wrappers": "error",
    "no-obj-calls": "error",
    "no-octal": "error",
    "no-octal-escape": "error",
    "no-path-concat": "error",
    "no-proto": "error",
    "no-redeclare": "error",
    "no-regex-spaces": "error",
    "no-return-assign": [
      "error",
      "except-parens"
    ],
    "no-return-await": "error",
    "no-self-assign": "error",
    "no-self-compare": "error",
    "no-sequences": "error",
    "no-shadow-restricted-names": "error",
    "no-sparse-arrays": "error",
    "no-tabs": "error",
    "no-template-curly-in-string": "error",
    "no-this-before-super": "error",
    "no-throw-literal": "error",
    "no-trailing-spaces": "error",
    "no-undef": "error",
    "no-undef-init": "error",
    "no-unexpected-multiline": "error",
    "no-unmodified-loop-condition": "error",
    "no-unneeded-ternary": [
      "error",
      {
        "defaultAssignment": false
      }
    ],
    "no-unreachable": "error",
    "no-unsafe-finally": "error",
    "no-unsafe-negation": "error",
    "no-unused-expressions": [
      "error",
      {
        "allowShortCircuit": true,
        "allowTernary": true,
        "allowTaggedTemplates": true
      }
    ],
    "no-unused-vars": [
      "error",
      {
        "vars": "all",
        "args": "none",
        "ignoreRestSiblings": true
      }
    ],
    "no-use-before-define": [
      "error",
      {
        "functions": false,
        "classes": false,
        "variables": false
      }
    ],
    "no-useless-call": "error",
    "no-useless-computed-key": "error",
    "no-useless-constructor": "error",
    "no-useless-escape": "error",
    "no-useless-rename": "error",
    "no-useless-return": "error",
    "no-whitespace-before-property": "error",
    "no-with": "error",
    "object-property-newline": [
      "error",
      {
        "allowMultiplePropertiesPerLine": true
      }
    ],
    "one-var": [
      "error",
      {
        "initialized": "never"
      }
    ],
    "operator-linebreak": [
      "error",
      "after",
      {
        "overrides": {
          "?": "before",
          ":": "before"
        }
      }
    ],
    "padded-blocks": [
      "error",
      {
        "blocks": "never",
        "switches": "never",
        "classes": "never"
      }
    ],
    "prefer-promise-reject-errors": "error",
    "quotes": [
      "error",
      "single",
      {
        "avoidEscape": true,
        "allowTemplateLiterals": true
      }
    ],
    "rest-spread-spacing": [
      "error",
      "never"
    ],
    "semi": [
      "error",
      "never"
    ],
    "semi-spacing": [
      "error",
      {
        "before": false,
        "after": true
      }
    ],
    "space-before-blocks": [
      "error",
      "always"
    ],
    "space-before-function-paren": [
      "error",
      "always"
    ],
    "space-in-parens": [
      "error",
      "never"
    ],
    "space-infix-ops": "error",
    "space-unary-ops": [
      "error",
      {
        "words": true,
        "nonwords": false
      }
    ],
    "spaced-comment": [
      "error",
      "always",
      {
        "line": {
          "markers": [
            "*package",
            "!",
            "/",
            ","
          ]
        },
        "block": {
          "balanced": true,
          "markers": [
            "*package",
            "!",
            ",",
            ":",
            "::",
            "flow-include"
          ],
          "exceptions": [
            "*"
          ]
        }
      }
    ],
    "symbol-description": "error",
    "template-curly-spacing": [
      "error",
      "never"
    ],
    "template-tag-spacing": [
      "error",
      "never"
    ],
    "unicode-bom": [
      "error",
      "never"
    ],
    "use-isnan": "error",
    "valid-typeof": [
      "error",
      {
        "requireStringLiterals": true
      }
    ],
    "wrap-iife": [
      "error",
      "any",
      {
        "functionPrototypeMethods": true
      }
    ],
    "yield-star-spacing": [
      "error",
      "both"
    ],
    "yoda": [
      "error",
      "never"
    ],
    "import/export": "error",
    "import/first": "error",
    "import/no-duplicates": "error",
    "import/no-webpack-loader-syntax": "error",
    "node/no-deprecated-api": "error",
    "node/process-exit-as-throw": "error",
    "promise/param-names": "error",
    "standard/array-bracket-even-spacing": [
      "error",
      "either"
    ],
    "standard/computed-property-even-spacing": [
      "error",
      "even"
    ],
    "standard/no-callback-literal": "error",
    "standard/object-curly-even-spacing": [
      "error",
      "either"
    ]
  },
  "parserOptions": {
    "ecmaVersion": 8,
    "ecmaFeatures": {
      "globalReturn": true,
      "experimentalObjectRestSpread": true,
      "jsx": true
    },
    "sourceType": "module"
  },
  "plugins": [
    "import",
    "node",
    "promise",
    "standard",
    "html"
  ],
  "root": true,
  "parser": "/home/dotnet/projects/napp/webviewer2/node_modules/babel-eslint/index.js",
  "extends": "standard"
}
```

</details>

**Comment by dotnetCarpenter** [comment]
I just updated to the latest version (4.11.0) and I have the same issue. Strange that the online demo seems to work with the same version.

```
npm i eslint@latest
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.1.3 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

+ eslint@4.11.0
added 117 packages, removed 2 packages and updated 2 packages in 56.385s
dotnet@dot-aspire:~/projects/napp/webviewer2 (tailwind)*$ grep version node_modules/eslint/package.json 
  "version": "4.11.0"
dotnet@dot-aspire:~/projects/napp/webviewer2 (tailwind)*$ node_modules/.bin/eslint assets/js/utils.js 

/home/dotnet/projects/napp/webviewer2/assets/js/utils.js
  68:30  error  Expected '===' and instead saw '=='  eqeqeq

✖ 1 problem (1 error, 0 warnings)
```

**Comment by dotnetCarpenter** [comment]
If I flip the values in the array (`'eqeqeq': ['smart', 'error']`) then I get:

>Configuration for rule "eqeqeq" is invalid:
>Severity should be one of the following: 0 = off, 1 = warn, 2 = error (you passed '"smart"').

**Comment by platinumazure** [comment]
Not going to lie, I'm stumped.

Are you sure you have the right version of ESLint? npm's deduplication logic can sometimes be weird- sometimes it will pull in a weird, old version of a dependency because that version is a dependency of a dependency... or something. Maybe worth double-checking with `eslint -v`, or doing an `npm ls | grep eslint` to see if there are other versions that could be lying around?

Maybe it might be worth `rm -rf ./node_modules` and then `npm install` again, just to make sure you have everything fresh?

Sorry I don't have any better ideas at this point. Honestly, from what I can tell with your config, everything should be working.

**Comment by platinumazure** [comment]
All rules have to start with a severity (0/1/2/off/warn/error) and then the rule options follow. So `["smart", "error"]` definitely isn't a valid configuration. On the other hand, `["error", "smart"]` definitely looked valid to me, and I checked the rule code.

**Comment by dotnetCarpenter** [comment]
The comment at the top of the file works, so I'll use that as a work-around. But pretty sure this is a bug in `Babel-ESLint`, since the rule is picked up by `eslint` according to the output of `--print-config`.

**Comment by dotnetCarpenter** [comment]
As you can see in my previous post, then grepping the version gives me:
```
grep version node_modules/eslint/package.json 
"version": "4.11.0"
```

I'll cleanup my previous post...

**Comment by dotnetCarpenter** [comment]
I did a `rm -r node_modules && npm i` and now I get a warning. I would expect nothing since it's a `null` check but it's better than before.

```
node_modules/.bin/eslint -c .eslintrc.js assets/js/utils.js 

assets/js/utils.js
  68:30  warning  Expected '===' and instead saw '=='  eqeqeq

✖ 1 problem (0 errors, 1 warning)
```

eslint version: 4.11.0

**Comment by dotnetCarpenter** [comment]
That's weird! I had `'eqeqeq': 1`when I did the previous test. If I change it back to `'eqeqeq': ['error', 'smart']` I get an error.

I'll leave `'eqeqeq': 1` and `/* eslint eqeqeq: [error, smart] */` at the top of `utils.js` for now...

**Comment by platinumazure** [comment]
Wait, where did you have `eqeqeq: 1`? Do you have multiple config files?

**Comment by dotnetCarpenter** [comment]
@platinumazure I did a search for `.eslintrc.js` files but nothing came up other than 2 modules in node_modules.

```
ls */**/.eslintrc.js
node_modules/domain-browser/.eslintrc.js	node_modules/hash.js/.eslintrc.js
```

I've tried to produce a reduced test case, https://github.com/dotnetCarpenter/eslint_issue, with a new nuxt installation but now, it seems that `.eslintrc.js` is not picked up at all.

+ [`.eslintrc.js`](https://github.com/dotnetCarpenter/eslint_issue/blob/master/.eslintrc.js)
+ [offensive file](https://github.com/dotnetCarpenter/eslint_issue/blob/master/assets/js/utils.js)

**Comment by platinumazure** [comment]
@dotnetCarpenter Apologies for letting this sit. Is this still an issue? If so, please post your latest configuration and ESLint version and I'll take another look.

**Comment by dotnetCarpenter** [comment]
I'll check again tonight.

**Comment by nzakas** [comment]
Unfortunately, it looks like there wasn't enough interest from the team or community to implement this change. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to be implemented after 90 days tend to never be implemented, and as such, we [close those issues](https://eslint.org/docs/maintainer-guide/issues#when-to-close-an-issue). This doesn't mean the idea isn't interesting or useful, just that it's not something the team can commit to.
